### PR TITLE
download error handling

### PR DESCRIPTION
**Pull request by dimacodota**

### CODE REVIEW

File: src/download.utils.ts [hunk 1]
```diff
@@ -60,9 +68,11 @@ export function downloadResource<T>(
           return resolve(downloadResource(redirectUrl, callback));
         }
         if (response.statusCode !== 200 && response.statusCode !== 403) {
-          return reject();
+          return reject(
+            new Error(`Failed request statusCode ${response.statusCode || ""}`)
```
Comment (boaz-codota) [inline src/download.utils.ts:72]: why new Error and not just a string?

File: src/test/runTest.ts [hunk 2]
```diff
@@ -30,29 +31,22 @@ async function main() {
...
-    if (err) {
-      console.error("Failed to copy test binaries", err);
-      process.exit(1);
-    }
-  });
+function copyTestBinaries(
+  fixtureBinary: string,
+  targetBinary: string
+): Promise<void> {
+  return promisify(ncp)(fixtureBinary, targetBinary);
```
Comment (boaz-codota) [inline src/test/runTest.ts:45]: extract `promisify(ncp)` plz

File: src/test/runTest.ts [hunk 3]
```diff
@@ -30,29 +31,22 @@ async function main() {
...
 
-function clearTestBinaries(targetBinary: string) {
-  rimraf(targetBinary, (err) => {
-    if (err) {
-      console.error("Failed to copy test binaries", err);
-      process.exit(1);
-    }
-  });
+function clearTestBinaries(targetBinary: string): Promise<void> {
+  return promisify(rimraf)(targetBinary);
```
Comment (boaz-codota) [inline src/test/runTest.ts:49]: plz extract `promisify(rimraf)`

**Review by boaz-codota — APPROVED** [review]
Cool

File: src/download.utils.ts (hunk 1 above)
Comment (dimacodota) [reply src/download.utils.ts:72]: error will include a stack trace

**Pull request merged**